src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/predicate/ValueEqualsKeywordsPredicate.java

```diff
@@ -0,0 +1,39 @@
+package ay2122s1_cs2103t_w16_2.btbb.model.predicate;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a value equals any of the keywords given.
+ *
+ * @param <T> Type of the predicate.
+ */
+public class ValueEqualsKeywordsPredicate<T, S> implements Predicate<T> {
+    private final Function<T, S> getter;
+    private final List<S> keywords;
+
+    /**
+     * Constructs a {@code ValueEqualsKeywordsPredicate}.
+     *
+     * @param getter Function to get the value to be tested.
+     * @param keywords List of keywords to be checked against.
+     */
+    public ValueEqualsKeywordsPredicate(Function<T, S> getter, List<S> keywords) {
+        this.getter = getter;
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(T t) {
+        S toTest = getter.apply(t);
+        return keywords.stream().anyMatch(toTest::equals);
+    }
```

Comment:
The point of this Predicate is to test if an object's field is equal to any of the values in a given list. Keywords is not the right term to use here because keywords tend to imply a String. Maybe rename it to 'values'. The name of the class should be like ValueInListPredicate. For your test method u can also just do:
```
return values.contains(getter.apply(t));
```